command/generate-root.go

```diff
@@ -0,0 +1,348 @@
+	var init, cancel, status, genotp bool
+	var nonce, decode, otp, pgpKey string
+	var pgpKeyArr pgpkeys.PubKeyFilesFlag
+	flags := c.Meta.FlagSet("generate-root", FlagSetDefault)
+	flags.BoolVar(&init, "init", false, "")
+	flags.BoolVar(&cancel, "cancel", false, "")
+	flags.BoolVar(&status, "status", false, "")
+	flags.BoolVar(&genotp, "genotp", false, "")
+	flags.StringVar(&decode, "decode", "", "")
+	flags.StringVar(&otp, "otp", "", "")
+	flags.StringVar(&nonce, "nonce", "", "")
+	flags.Var(&pgpKeyArr, "pgp-key", "")
+	flags.Usage = func() { c.Ui.Error(c.Help()) }
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	if genotp {
+		buf := make([]byte, 16)
+		readLen, err := rand.Read(buf)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error reading random bytes: %s", err))
+			return 1
+		}
+		if readLen != 16 {
+			c.Ui.Error(fmt.Sprintf("Read %d bytes when we should have read 16", readLen))
+			return 1
+		}
+		c.Ui.Output(fmt.Sprintf("OTP: %s", base64.StdEncoding.EncodeToString(buf)))
+		return 0
+	}
+
+	if len(decode) > 0 {
+		if len(otp) == 0 {
+			c.Ui.Error("Both the value to decode and the OTP must be passed in")
+			return 1
+		}
+		return c.decode(decode, otp)
+	}
+
+	client, err := c.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf(
+			"Error initializing client: %s", err))
+		return 2
+	}
+
+	// Check if the root generation is started
+	rootGenerationStatus, err := client.Sys().RootGenerationStatus()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error reading root generation status: %s", err))
+		return 1
+	}
+
+	// If we are initing, or if we are not started but are not running a
+	// special function, check otp and pgpkey
+	if init ||
+		(!init && !cancel && !status && !genotp && len(decode) == 0 && !rootGenerationStatus.Started) {
+		switch {
+		case len(otp) == 0 && (pgpKeyArr == nil || len(pgpKeyArr) == 0):
+			c.Ui.Error("-otp or -pgp-key must be specified")
+			return 1
+		case len(otp) != 0 && pgpKeyArr != nil && len(pgpKeyArr) != 0:
+			c.Ui.Error("Only one of -otp or -pgp-key must be specified")
+			return 1
+		case len(otp) != 0:
+			err := c.verifyOTP(otp)
+			if err != nil {
+				c.Ui.Error(fmt.Sprintf("Error verifying the provided OTP: %s", err))
+				return 1
+			}
+		case pgpKeyArr != nil:
+			if len(pgpKeyArr) != 1 {
+				c.Ui.Error("Could not parse PGP key")
+				return 1
+			}
+			if len(pgpKeyArr[0]) == 0 {
+				c.Ui.Error("Got an empty PGP key")
+				return 1
+			}
+			pgpKey = pgpKeyArr[0]
+		default:
+			panic("unreachable case")
+		}
+	}
+
+	if nonce != "" {
+		c.Nonce = nonce
+	}
+
+	// Check if we are running doing any restricted variants
+	switch {
+	case init:
+		return c.initRootGeneration(client, otp, pgpKey)
+	case cancel:
+		return c.cancelRootGeneration(client)
+	case status:
+		return c.rootGenerationStatus(client)
+	}
+
+	// Start the root generation process if not started
+	if !rootGenerationStatus.Started {
+		err = client.Sys().RootGenerationInit(otp, pgpKey)
```

Comment:
It actually basically buys nothing because then you have to take the output, check the errors, and then decide whether to return with an int or not. So I'll leave it...it's only one call/error check that's duplicated.